k8-operator/controllers/infisicalsecret_controller.go

```diff
@@ -187,12 +187,18 @@ func (r *InfisicalSecretReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 func (r *InfisicalSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&secretsv1alpha1.InfisicalSecret{}).
+		// For the Infisical Secret CRD we specifically watch for the update event, because we need to delete the entry in the resourceVariablesMap, so it will refresh the variables used to fetch secrets.
+		For(&secretsv1alpha1.InfisicalSecret{}, builder.WithPredicates(predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				delete(resourceVariablesMap, string(e.ObjectNew.GetUID()))
```

Comment:
So deleting this will trigger CRD update? (not a big fan of the finalizer pattern, i have seen customers run into issues where the delete is in a stuck state, but this is unrelated to the PR)